src/ca65/ea65.c

```diff
@@ -140,16 +140,27 @@ void GetEA (EffAddr* A)
 
         } else {
 
-            /* (adr) or (adr),y */
+            /* (adr), (adr),y or (adr),z */
             Consume (IndirectLeave, IndirectExpect);
             if (CurTok.Tok == TOK_COMMA) {
                 /* (adr),y */
                 NextTok ();
-                Consume (TOK_Y, "`Y' expected");
-                A->AddrModeSet = AM65_DIR_IND_Y;
+                switch(CurTok.Tok) {
+                case TOK_Z:
+                    if (CPU == CPU_4510) {
+                        NextTok ();
+                        A->AddrModeSet = AM65_DIR_IND;
+                    }
+                    break;
```

Comment:
I think that the `break;` should be removed.  Then, if someone does use ",z" on a non-4510 CPU, ca65 will say that a "y" should have been used.